src/libs/shim-interface/src/lib.rs

```diff
@@ -15,16 +15,57 @@
 //! HTTP request to the server. The server inside shim will multiplex the request
 //! to its corresponding handler and run certain methods.
 
-use std::path::Path;
+use std::fs;
 
-use anyhow::{anyhow, Result};
+use anyhow::{anyhow, Context, Result};
 
 pub mod shim_mgmt;
 
 use kata_types::config::KATA_PATH;
 
 pub const SHIM_MGMT_SOCK_NAME: &str = "shim-monitor.sock";
 
+fn get_uds_with_sid(short_id: &str, path: &str) -> Result<String> {
+    // /kata/run/62db246d6c4ffc4a20b56b7259a0346a78aeccb46c4bdc3ef97899d787fd5797
+    let kata_run_path = fs::canonicalize(path).context("It is not a canonical, absolute path")?;
```

Comment:
Please could you add a call to `kata_sys_util::validate::verify_id()` here (`Cargo.toml` update required) to ensure we validate that the ID doesn't contain any path elements:

```suggestion
fn get_uds_with_sid(short_id: &str, path: &str) -> Result<String> {
    verify_id(short_id)?;

    // /kata/run/62db246d6c4ffc4a20b56b7259a0346a78aeccb46c4bdc3ef97899d787fd5797
    let kata_run_path = fs::canonicalize(path).context("It is not a canonical, absolute path")?;
```

Since it's dealing with paths, for safety I'd suggest adding a few unit tests for this function to assert the expected behaviour (pass in blank/invalid `short_id`, `short_id` with zero, 1, multiple matches, _etc_ and use a blank/tmpdir value for `path`, _etc_).

Reply:
Yeah,Thx James, good suggestion! And I will add UT

Reply:
Hi James @jodh-intel I have updated my code. Could you please review it again. Thx